src/components/Input/searchInput.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import { SearchIcon } from '../../assets';
+import IconWrap from '../../components/ui/svgWrapper';
+import { useAppDispatch, useAppSelector } from '../../redux/hooks';
+import { setSearch } from '../../redux/slice/querySearch';
+import useDebounce from '../hooks';
+const SearchInput = () => {
+  const [value, setValue] = useState('');
+  const { search } = useAppSelector((state) => state.query);
+  const dispatch = useAppDispatch();
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  const debouncedValue = useDebounce(value, 500);
+
+  const searchedValue = useCallback(() => {
+    dispatch(setSearch({ ...search, search_txt: debouncedValue }));
+  }, [debouncedValue]);
+
+  useEffect(() => {
+    searchedValue();
+  }, [debouncedValue, value]);
+
```

Comment:
All this is unnecessary

Reply:
should I do away with the useCallback? It was used so that the function does not run anytime there is a re-render